src/shell-app.c

```diff
@@ -937,7 +939,10 @@ shell_app_sync_running_state (ShellApp *app)
 
   if (app->state != SHELL_APP_STATE_STARTING)
     {
-      if (app->running_state->interesting_windows == 0)
+      /* We check if the startup completed as a workaround for apps that show
+       * a window and quickly remove it before the main window shows up, which
+       * is the case with Maps. */
+      if (app->running_state->interesting_windows == 0 && app->running_state->startup_complete)
```

Comment:
I tried to avoid going through all the windows. I know there shouldn't be that much but I try to make my code optimal (yes, even if I've wrote the IO thingy yesterday :D ).